### PR TITLE
chore: repo maintenance + no path dependencies

### PR DESCRIPTION
**Pull request by carllerche**
- Move `tokio` into its own directory.
- Remove `path` dependencies.
- Run CI once with `crates.io` dependencies and one with path dependencies.

The `examples` directory still has to be moved.

### CODE REVIEW

**Comment by carllerche** [comment]
I moved the `tokio` crate to be more consistent (having a random `src` / `tests` directory at the root was weird I think). Also, it required a bunch of special casing in the CI scripts.

File: tokio/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+uds = ["tokio-uds"]
+
+# This feature comes with no promise of stability. Things will
+# break with each patch release. Use at your own risk.
+async-await-preview = [
+  "tokio-async-await/async-await-preview",
+]
+
+[badges]
+travis-ci = { repository = "tokio-rs/tokio" }
```
Comment (LucioFranco) [inline tokio/Cargo.toml]: also we could prob remove these badges.

Comment (carllerche) [reply tokio/Cargo.toml]: Done

**Comment by hawkw** [comment]
@carllerche I believe the change moving `tokio` into its own directory broke the examples, since they are no longer part of a crate cargo knows about. I'll open a PR fixing this.

**Comment by carllerche** [comment]
thanks 👍

**Pull request merged**